### PR TITLE
Exceptions fix - merge scipr-lab/libfqfft#12 (depends on #6)

### PR DESCRIPTION
**Pull request by dtebbs**
Merge @HarryR's PR to remove the use of exceptions in domain selection logic:  scipr-lab/libfqfft#12
Original branch: https://github.com/Ethsnarks/libfqfft/tree/no-exceptional-control-flow

### CODE REVIEW

**Comment by dtebbs** [comment]
Relies on clearmatics/libff#64

File: libfqfft/evaluation_domain/domains/arithmetic_sequence_domain.tcc [hunk 1]
```diff
@@ -23,6 +23,20 @@
 
 namespace libfqfft {
 
+template<typename FieldT>
+bool arithmetic_sequence_domain<FieldT>::valid_for_size(const size_t m)
+{
+  if (m <=1) {
```
Comment (AntoineRondelet) [inline libfqfft/evaluation_domain/domains/arithmetic_sequence_domain.tcc]: `m <=1` -> `m <= 1`

File: libfqfft/evaluation_domain/domains/geometric_sequence_domain.tcc [hunk 2]
```diff
@@ -23,19 +23,33 @@
 
 namespace libfqfft {
 
+template<typename FieldT>
+bool geometric_sequence_domain<FieldT>::valid_for_size(const size_t m)
+{
+  if ( m <= 1 ) {
```
Comment (AntoineRondelet) [inline libfqfft/evaluation_domain/domains/geometric_sequence_domain.tcc]: `if ( m <= 1 ) {` -> `if (m <= 1) {`

File: libfqfft/evaluation_domain/domains/arithmetic_sequence_domain.tcc (hunk 1 above)
Comment (dtebbs) [reply libfqfft/evaluation_domain/domains/arithmetic_sequence_domain.tcc]: Should be fixed now.  Thanks.

File: libfqfft/evaluation_domain/domains/geometric_sequence_domain.tcc (hunk 2 above)
Comment (dtebbs) [reply libfqfft/evaluation_domain/domains/geometric_sequence_domain.tcc]: Thanks. I must have missed this one.  Fixed.

**Review by AntoineRondelet — APPROVED** [review]
LGTM

**Pull request merged**